### PR TITLE
fix: improve abortion check to use nats events

### PR DESCRIPTION
**Pull request by nicufk**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: internal/app/api/v1/testsuites.go [hunk 1]
```diff
@@ -803,11 +804,14 @@ func (s TestkubeAPI) AbortTestSuiteHandler() fiber.Handler {
 
 		for _, execution := range executions {
 			execution.Status = testkube.TestSuiteExecutionStatusAborting
-			err = s.TestExecutionResults.Update(c.Context(), execution)
+			s.Log.Infow("aborting test suite execution", "executionID", execution.Id)
+			err := bus.Internal.PublishTopic(bus.InternalPublishTopic, testkube.NewEventEndTestSuiteAborted(&execution))
```
Comment (vsukhin) [inline internal/app/api/v1/testsuites.go:808]: I would store it in Server object, not global var

File: pkg/event/bus/nats.go [hunk 2]
```diff
@@ -36,9 +41,10 @@ func NewNATSConnection(uri string) (*nats.EncodedConn, error) {
 }
 
 func NewNATSBus(nc *nats.EncodedConn) *NATSBus {
-	return &NATSBus{
+	Internal = &NATSBus{
```
Comment (vsukhin) [inline pkg/event/bus/nats.go:44]: I'm trying to avoid global vars

File: pkg/scheduler/testsuite_scheduler.go [hunk 3]
```diff
@@ -106,85 +108,102 @@ func (s *Scheduler) runSteps(ctx context.Context, wg *sync.WaitGroup, testsuiteE
...
 
-	go s.abortionCheck(ctx, testsuiteExecution, request.Timeout, abortChan)
+	go s.timeoutCheck(ctx, testsuiteExecution, request.Timeout)
 
-	for i := range testsuiteExecution.ExecuteStepResults {
-		batchStepResult = &testsuiteExecution.ExecuteStepResults[i]
-		select {
-		case abortionStatus = <-abortChan:
-			s.logger.Infow("Aborting test suite execution", "execution", testsuiteExecution.Id, "i", i)
+	err := bus.Internal.SubscribeTopic(bus.InternalSubscribeTopic, testsuiteExecution.Name, func(event testkube.Event) error {
```
Comment (vsukhin) [inline pkg/scheduler/testsuite_scheduler.go:119]: I would prefee to store it in Schedule object

File: pkg/scheduler/testsuite_scheduler.go [hunk 4]
```diff
@@ -106,85 +108,102 @@ func (s *Scheduler) runSteps(ctx context.Context, wg *sync.WaitGroup, testsuiteE
...
+		s.logger.Errorw("error subscribing to event", "error", err)
+	}
 
-				continue
-			}
+	for i := range testsuiteExecution.ExecuteStepResults {
+		batchStepResult = &testsuiteExecution.ExecuteStepResults[i]
+		s.logger.Debugw("Running batch step", "step", batchStepResult.Execute, "i", i)
+
+		if cancelSteps.Load() {
```
Comment (vsukhin) [inline pkg/scheduler/testsuite_scheduler.go:153]: Does this code  depends from SubscribeTopic? It should be async, no?

Comment (vsukhin) [reply pkg/scheduler/testsuite_scheduler.go:153]: Do we wait as in code below/

Comment (nicufk) [reply pkg/scheduler/testsuite_scheduler.go]: fixed, it uses channels to sync and handler has no logic in it

**Review by vsukhin — APPROVED** [review]
looks good

**Pull request merged**